src/clr-angular/data/datagrid/render/header-renderer.ts

```diff
@@ -4,16 +4,33 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 import { Directive, ElementRef, EventEmitter, OnDestroy, Output, Renderer2 } from '@angular/core';
-import { Subscription } from 'rxjs';
+import { BehaviorSubject, Subscription } from 'rxjs';
 
 import { DomAdapter } from '../../../utils/dom-adapter/dom-adapter';
 import { DatagridRenderStep } from '../enums/render-step.enum';
 import { ColumnResizerService } from '../providers/column-resizer.service';
 import { STRICT_WIDTH_CLASS } from './constants';
 import { DatagridRenderOrganizer } from './render-organizer';
+import { DatagridColumnState } from '../interfaces/column-state.interface';
+import { DatagridColumnChanges } from '../enums/column-changes.enum';
 
 @Directive({ selector: 'clr-dg-column', providers: [ColumnResizerService] })
 export class DatagridHeaderRenderer implements OnDestroy {
+  private stateSubscription: Subscription;
+  private _columnState: BehaviorSubject<DatagridColumnState>;
+
+  get columnState(): BehaviorSubject<DatagridColumnState> {
+    return this._columnState;
+  }
+
+  set columnState(columnState: BehaviorSubject<DatagridColumnState>) {
```

Comment:
In dynamic columns, if we toggle additional columns, we will run this method for every column. Should we have a guard to check their references, exit out early if the `columnState` argument has the same reference as whatever saved to `_columnState`?

Reply:
When columns change, we do need to run this method to ensure they sync their latest states. If a new column arrives it might change the widths of other columns (such as last column has the flex column, so another one could take on a specific width defined). The frequency of which columns change is not a concern in performance, as the rows are more likely to change with pagination or data refreshing.